### PR TITLE
#213: Fix uploading csv files from Windows

### PR DESCRIPTION
**Pull request by kcissewski**

### CODE REVIEW

File: src/main/webapp/app/modules/upload/upload-page.tsx [hunk 1]
```diff
@@ -109,9 +109,21 @@ export class UploadPage extends React.Component<IUploadPageProp, IUploadState> {
...
+      type = this.handleUndefinedFileType(source, type);
       resolve(type);
     });
 
+  handleUndefinedFileType = (source, type) => {
+    if (type === '') {
+      const re = /(?:\.([^.]+))?$/;
+      const fileExtension = re.exec(source.name)[1];
+      if (fileExtension === 'csv') {
+        type = 'text/csv';
```
Comment (wojtekbu) [inline src/main/webapp/app/modules/upload/upload-page.tsx]: Can only CSV be undefined?

File: src/main/webapp/app/modules/upload/upload-page.tsx [hunk 2]
```diff
@@ -109,9 +109,21 @@ export class UploadPage extends React.Component<IUploadPageProp, IUploadState> {
 
   fileValidateTypeDetectType = (source, type) =>
     new Promise((resolve, reject) => {
+      type = this.handleUndefinedFileType(source, type);
       resolve(type);
     });
 
+  handleUndefinedFileType = (source, type) => {
+    if (type === '') {
+      const re = /(?:\.([^.]+))?$/;
```
Comment (wojtekbu) [inline src/main/webapp/app/modules/upload/upload-page.tsx]: Could you describe this regex by changing its name?

File: src/main/webapp/app/modules/upload/upload-page.tsx [hunk 3]
```diff
@@ -109,9 +109,21 @@ export class UploadPage extends React.Component<IUploadPageProp, IUploadState> {
 
   fileValidateTypeDetectType = (source, type) =>
     new Promise((resolve, reject) => {
+      type = this.handleUndefinedFileType(source, type);
       resolve(type);
     });
 
+  handleUndefinedFileType = (source, type) => {
```
Comment (wojtekbu) [inline src/main/webapp/app/modules/upload/upload-page.tsx]: Does it handle undefined on any system/browser or for a specific one?

File: src/main/webapp/app/modules/upload/upload-page.tsx (hunk 1 above)
Comment (kcissewski) [reply src/main/webapp/app/modules/upload/upload-page.tsx]: As far as I know in our case only CSV file may cause problems.

File: src/main/webapp/app/modules/upload/upload-page.tsx (hunk 2 above)
Comment (kcissewski) [reply src/main/webapp/app/modules/upload/upload-page.tsx]: Right, I forgot about it, thanks!

File: src/main/webapp/app/modules/upload/upload-page.tsx (hunk 3 above)
Comment (kcissewski) [reply src/main/webapp/app/modules/upload/upload-page.tsx]: It handles every system/browser.

**Pull request merged**